### PR TITLE
Assessments:  Rename tables and add comments

### PR DESCRIPTION
**Pull request by nkiruka**
The tables and prop types in the assessments tab are renamed to be more descriptive and reflect what the table is used for.  Comments are also added for code clarity.

See previous PR [22885](https://github.com/code-dot-org/code-dot-org/pull/22885) for reference.

### CODE REVIEW

File: apps/src/templates/sectionAssessments/MultipleChoiceOverviewTable.jsx [hunk 1]
```diff
@@ -57,16 +57,19 @@ const answerDataPropType = PropTypes.shape({
...
+const multipleChoiceDataPropType = PropTypes.shape({
   id: PropTypes.number.isRequired,
   question: PropTypes.string.isRequired,
   answers: PropTypes.arrayOf(answerDataPropType),
   notAnswered: PropTypes.number.isRequired,
 });
 
+// A single table that shows students' responses to each multiple choice question.
+// The table displays the percent of students that select an answer choice and
+// percent of students that did not answer the question.
```
Comment (maddiedierker) [inline apps/src/templates/sectionAssessments/MultipleChoiceOverviewTable.jsx]: nit: multi-line comments in js should look like this:
```
/**
 * A single table that shows students' responses to each multiple choice question.
 * The table displays the percent of students that select an answer choice and
 * percent of students that did not answer the question.
 */
```

File: apps/src/templates/sectionAssessments/StudentsAssessmentsMCTable.jsx [hunk 2]
```diff
@@ -59,7 +59,13 @@ const studentOverviewDataPropType = PropTypes.shape({
   submissionStatus: PropTypes.string,
 });
 
-class StudentAssessmentOverviewTable extends Component {
+// A table that shows the summary data for each student in a section.
+// Each row is a single student, the number of questions the student
+// answered correctly, the total number of multiple choice questions,
+// the precent of correct answers, and status of each student's
+// assessment or a time-stamp for when a student submits an
+// assessment.
```
Comment (maddiedierker) [inline apps/src/templates/sectionAssessments/StudentsAssessmentsMCTable.jsx]: same multi-line js comment nit as above

File: apps/src/templates/sectionAssessments/MultipleChoiceByStudentContainer.jsx [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 } from './sectionAssessmentsRedux';
 import { connect } from 'react-redux';
 
-class MultipleChoiceByStudentSection extends Component {
+class MultipleChoiceByStudentContainer extends Component {
   static propTypes = {
-    multipleChoiceStructure: PropTypes.arrayOf(questionStructurePropType),
-    studentAnswerData: PropTypes.arrayOf(studentAnswerDataPropType),
+    multipleChoiceStructure: PropTypes.arrayOf(multipleChoiceQuestionPropType),
+    studentAnswerData: PropTypes.arrayOf(studentWithResponsesPropType),
```
Comment (maddiedierker) [inline apps/src/templates/sectionAssessments/MultipleChoiceByStudentContainer.jsx:13]: 👍 love the naming changes on 12 and 13

File: apps/src/templates/sectionAssessments/assessmentsTestHelpers.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import i18n from '@cdo/locale';
 
+// Data for students assessments multiple choice table
```
Comment (maddiedierker) [inline apps/src/templates/sectionAssessments/assessmentsTestHelpers.js]: is this data for multiple students? if it is, this should say "Data for _students'_ assessments multiple choice table"

File: apps/src/templates/sectionAssessments/assessmentDataShapes.js [hunk 5]
```diff
@@ -15,23 +15,23 @@ export const questionDataPropType = PropTypes.shape({
 
 // Shapes for single student multiple choice tables
 
-// Represents an individual students response to a single
+// Represents an individual student's response to a single
 // multiple choice question.
 export const studentResponsePropType = PropTypes.shape({
   isCorrect: PropTypes.bool,
-  answers: PropTypes.string,
+  responses: PropTypes.string,
```
Comment (caleybrock) [inline apps/src/templates/sectionAssessments/assessmentDataShapes.js]: Might be good to have a comment here with what these strings could look like: `A` or `A, C`, etc.

**Comment by caleybrock** [comment]
Looks like there are some legitimate test failures circle is finding as well.

Overall though, this is super helpful!

File: apps/src/templates/sectionAssessments/MultipleChoiceOverviewTable.jsx (hunk 1 above)
Comment (nkiruka) [reply apps/src/templates/sectionAssessments/MultipleChoiceOverviewTable.jsx]: Fixed.

File: apps/src/templates/sectionAssessments/StudentsAssessmentsMCTable.jsx (hunk 2 above)
Comment (nkiruka) [reply apps/src/templates/sectionAssessments/StudentsAssessmentsMCTable.jsx]: Fixed, Thanks!

File: apps/src/templates/sectionAssessments/assessmentsTestHelpers.js (hunk 4 above)
Comment (nkiruka) [reply apps/src/templates/sectionAssessments/assessmentsTestHelpers.js]: Fixed.  Thanks

File: apps/src/templates/sectionAssessments/MultipleChoiceOverviewTable.jsx [hunk 6]
```diff
@@ -57,16 +57,20 @@ const answerDataPropType = PropTypes.shape({
...
 
-const questionDataPropType = PropTypes.shape({
+const multipleChoiceDataPropType = PropTypes.shape({
   id: PropTypes.number.isRequired,
   question: PropTypes.string.isRequired,
   answers: PropTypes.arrayOf(answerDataPropType),
   notAnswered: PropTypes.number.isRequired,
 });
 
+/* A single table that shows students' responses to each multiple choice question.
```
Comment (caleybrock) [inline apps/src/templates/sectionAssessments/MultipleChoiceOverviewTable.jsx]: your multi line comments are missing some `*`s. 

Check out `starred-block` in this link: https://eslint.org/docs/rules/multiline-comment-style

**Comment by caleybrock** [comment]
Really excited about this change!!

**Pull request merged**